File: LLM-Translations/GPT-4/avatar/Python/C/codeforces_272_B.c
#include <stdio.h>
#include <stdlib.h>

int f(int n) {
    int y = 1;
    while(n != 1) {
        if(n % 2) {
            y += 1;
        }
        n /= 2;
    }
    return y;
}

int nPairsWithCommonFX(int* sequence, int sequence_length) {
    int *storage[100];
    int FXs[100];
    int FXs_length = 0;
    for(int i = 0; i < sequence_length; i++) {
        int y = f(sequence[i]);
        int found = 0;
        for(int j = 0; j < FXs_length; j++) {
            if(FXs[j] == y) {
                found = 1;
                break;
            }
        }
        if(!found) {
            storage[y] = (int*) malloc(sizeof(int));
            storage[y][0] = sequence[i];
            FXs[FXs_length++] = y;
        } else {
            int storage_length = 0;
            while(storage[y][storage_length]) {
                storage_length++;
            }
            storage[y][storage_length] = sequence[i];
        }
    }
    int sum1 = 0, sum2 = 0;
    for(int i = 0; i < FXs_length; i++) {
        int y = FXs[i];
        int storage_length = 0;
        while(storage[y][storage_length]) {
            storage_length++;
        }
        sum1 += storage_length * storage_length;
        sum2 += storage_length;
    }
    return (sum1 - sum2) / 2;
}

int main() {
    int sequence_length;
    scanf("%d", &sequence_length);
    int sequence[sequence_length];
    for(int i = 0; i < sequence_length; i++) {
        scanf("%d", &sequence[i]);
    }
    printf("%d\n", nPairsWithCommonFX(sequence, sequence_length));
    return 0;
}

/* End of Code */

File: LLM-Translations/GPT-4/avatar/Python/Java/codeforces_272_B.java
import java.util.*;

public class codeforces_272_B {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        sc.nextLine();
        int[] sequence = Arrays.stream(sc.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        System.out.println(nPairsWithCommonFX(sequence));
    }

    public static int nPairsWithCommonFX(int[] sequence) {
        Map<Integer, List<Integer>> storage = new HashMap<>();
        List<Integer> FXs = new ArrayList<>();

        for (int value : sequence) {
            int y = f(value);
            if (!storage.containsKey(y)) {
                storage.put(y, new ArrayList<>(Arrays.asList(value)));
                FXs.add(y);
            } else {
                storage.get(y).add(value);
            }
        }

        int sum1 = 0;
        int sum2 = 0;
        for (int y : FXs) {
            sum1 += storage.get(y).size() * storage.get(y).size();
            sum2 += storage.get(y).size();
        }

        return (sum1 - sum2) / 2;
    }

    public static int f(int n) {
        int y = 1;
        while (n != 1) {
            if (n % 2 != 0) {
                y += 1;
            }
            n /= 2;
        }
        return y;
    }
}
// End of Code
